app/flatpak-builtins-build-commit-from.c

```diff
@@ -573,6 +575,9 @@ flatpak_builtin_build_commit_from (int argc, char **argv, GCancellable *cancella
               strcmp (key, OSTREE_COMMIT_META_KEY_ENDOFLIFE_REBASE) == 0)
             continue;
 
+          if (opt_need_token >= 0 && strcmp (key, "xa.need-token") == 0)
```

Comment:
Yeah, token-type seem better. However, the domain of the metadata is guint32, it just is currently using a signed int in the tools that set it so that i can use < 0 to mean "don't set", because I want to be able to signal unset to mean "whatever is default for this repo" which we then can explicitly override with "0" if we want the default to be require token with some refs not requiring it.

Reply:
I guess we *could* use int32 in the file too, and allow an explicit -1 to mean use the default. Might make things easier.

Reply:
Yeah, I’d err towards making the default an explicit value (rather than half of the domain, `< 0`) and adding an enum somewhere:
```
typedef enum
{
  FLATPAK_COMMIT_TOKEN_REPO_DEFAULT = -1,
  FLATPAK_COMMIT_TOKEN_NONE = 0,
  FLATPAK_COMMIT_TOKEN_PURCHASE = 1,
} FlatpakCommitTokenType;
```
or something.